src/lib/coda_networking/coda_networking.ml

```diff
@@ -749,9 +749,44 @@ module Make (Inputs : Inputs_intf) = struct
     let online_status =
       online_broadcaster config.time_controller online_notifier
     in
+    let filtered_gossips_reader, filtered_gossips_writer =
+      Strict_pipe.create ~name:"filtered_gossips"
+        (Buffered (`Capacity 50, `Overflow Crash))
+    in
+    ( if Option.is_none config.gossip_net_params.filter_layer then
+      Strict_pipe.transfer received_gossips filtered_gossips_writer ~f:Fn.id
+      |> don't_wait_for
+    else
+      let filter_program =
+        Option.value_exn config.gossip_net_params.filter_layer
+      in
+      don't_wait_for
+      @@ match%bind Process.create ~prog:filter_program ~args:[] () with
+         | Ok filter_process ->
+             Linear_pipe.transfer
+               (received_gossips |> Strict_pipe.Reader.to_linear_pipe)
+               (Process.stdin filter_process |> Writer.pipe)
+               ~f:(fun gossip ->
+                 Envelope.Incoming.sexp_of_t Message.sexp_of_msg gossip
```

Comment:
These should really be yojson or Joel will be sad.

Reply:
yojson is used for display-pretty-printing and thus we hide some of the data :( . We do need to fix this at some point, but for now we'll probably need to suffer with sexp

Reply:
if the filter process is logging the gossip, wouldn't it be better to use yojson here, to be consistent with the Coda logs?